### ISSUE TITLE
Custom login field for unitymedia Connect box not selectable

### ISSUE THREAD

**Issue by pstorch** [keepassxreboot/keepassxc-browser#360]
Just got a new modem from unitymedia (Connect Box). There is only a password field to login. It is not recognized as password field and can't be selected as custom login field.

## Expected Behavior
It should be possible to select the password field.

## Current Behavior
The single field on the login page is only selectable as username field, but not as password field.

###
The password field looks like this:
```
<input type="text" id="loginPassword" name="loginPassword" value="" class="default" onblur="InputTextDefault('loginPassword','');" onfocus="InputTextEnable('loginPassword','');$('#PasswordStrBar').show(); checkPskey();" maxlength="31" autofocus="" data-cip-id="loginPassword">
```

## Debug info
KeePassXC - 2.3.4
KeePassXC-Browser - 1.3.0
Operating system: Linux
Browser: Firefox

**Comment by droidmonkey** [comment]
That is because it is not a password field. This is some lunatic web design going on here.

**Comment by pstorch** [comment]
> That is because it is not a password field. This is some lunatic web design going on here.

That's correct. It shows the password as you type it :man_facepalming: 

> You are going to have to right click -> KeePassXC-Browser -> Fill Password Only
Already tried it:

![image](https://user-images.githubusercontent.com/5598124/50551196-551e3880-0c7d-11e9-9e35-cc74b955986c.png)

**Comment by droidmonkey** [comment]
OK that is a bug, when the user specifically requests a password filled on a form element we should just do it.

**Comment by varjolintu** [comment]
That is not a bug. Input fields are handled as combinations which can include username field, password field or both. This "password" field has input type `text` which is very bad design, as it can be identified as username field (or a generic search field etc.).

You can try to select `Choose custom credential fields` from the popup and skip the username field part, then select the field as a password field. This could do the trick.

**Comment by pstorch** [comment]
> That is not a bug. Input fields are handled as combinations which can include username field, password field or both. This "password" field has input type `text` which is very bad design, as it can be identified as username field (or a generic search field etc.).

I'm fine with closing this issue with `won't fix`, because this login page is misusing a text field as password field.

> You can try to select `Choose custom credential fields` from the popup and skip the username field part, then select the field as a password field. This could do the trick.

That's the problem I tried to described in the issue description. It only offers the password field as username field. After skipping the username, the password field is not selectable.

![image](https://user-images.githubusercontent.com/5598124/50564646-e5fd1e80-0d26-11e9-880d-bfd52f48637d.png)

![image](https://user-images.githubusercontent.com/5598124/50564652-f1e8e080-0d26-11e9-93ec-254e0310df96.png)

**Comment by varjolintu** [comment]
One hacky way you could try is using the TOTP filling as password fill, if it's possible to select the field as String Field? Then just add to the following to the entry attributes: `KPH: {TOTP}` but for the value, set `{PASSWORD}`.

**Comment by pstorch** [comment]
I just had a little success with KeePassXC Auto-Type if I name the entry `Firefox`, but this is also not ideal. Right-clicking in the field and choose Fill password into the selected field would be awesome. 

While I was typing this response there was this new tip from @varjolintu coming in, so I experimented a bit further and found a solution: 

I selected only the username field and skipped the password field. Then I edited the storage.js file and swapped username and password from
`{"username":"loginPassword","password":null,"fields":[]}` to `{"username":null,"password":"loginPassword","fields":[]}`

Now it works for me :smile:

**Comment by droidmonkey** [comment]
Perhaps we should allow selection of any field as the password field

**Comment by pstorch** [comment]
I was interested in browser extension development. Never done it before. So I looked into the idea of @droidmonkey:

Added a `Show more` button:
![image](https://user-images.githubusercontent.com/5598124/50572980-fe267980-0dca-11e9-969d-2458b76a8b4e.png)

After clicking the new button, the text field is selectable:
![image](https://user-images.githubusercontent.com/5598124/50572983-10081c80-0dcb-11e9-99b5-d899b9a135fd.png)

If you like it's worth it, I can send a PR. Otherwise it might still serve as a basis for further discussion.
